### PR TITLE
libbitcoin-explorer 3.3.0 (new formula)

### PR DESCRIPTION
**Pull request by sugarjig**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Add formula for libbitcoin-explorer, a command-line tool for Bitcoin (https://github.com/libbitcoin/libbitcoin-explorer). The formula specifies secp256k1, libbitcoin-protocol, libbitcoin-network, and libbitcoin-client as resources to build first. These repositories do not meet the required number of forks, watchers, and stars for creating separate formulas.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@sugarjig it's still showing "First-time contributor" for you. It seems the author of the commit is somehow disconnected from your GitHub account. It would be great if you could fix that so that our contribution statistics are tabulated correctly.

**Comment by sugarjig** [comment]
@ilovezfs Woops! Sorry about that. All fixed now.

File: Formula/libbitcoin-explorer.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "zeromq"
+  depends_on "libbitcoin"
+
+  resource "secp256k1" do
+    url "https://github.com/libbitcoin/secp256k1/archive/v0.1.0.13.tar.gz"
```
Comment (ilovezfs) [inline Formula/libbitcoin-explorer.rb]: Will the libbitcoin/secp256k1 fork be creating new tags here regularly or should we just be tying these to the upstream repository with a specific revision?

For example,
```
  resource "secp256k1" do
    url "https://github.com/bitcoin-core/secp256k1.git",
        :revision => "9d560f992db26612ce2630b194aef5f44d63a530"
  end
```

File: Formula/libbitcoin-explorer.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    system "./autogen.sh"
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/bx", "seed"
```
Comment (ilovezfs) [inline Formula/libbitcoin-explorer.rb]: Can we get this to verify something successful happened other than just a zero exit code? Perhaps using an assert?

**Comment by ilovezfs** [comment]
@sugarjig any news here?

File: Formula/libbitcoin-explorer.rb (hunk 1 above)
Comment (sugarjig) [inline Formula/libbitcoin-explorer.rb]: I want to say that the fork will be creating tags regularly. The libbitcoin team has a much better idea of which exact version to use. Is there a reason to prefer the upstream repo?

File: Formula/libbitcoin-explorer.rb (hunk 2 above)
Comment (sugarjig) [inline Formula/libbitcoin-explorer.rb]: Is there a good example I can look at? I'm a Ruby n00b...

File: Formula/libbitcoin-explorer.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/libbitcoin-explorer.rb]: >Is there a reason to prefer the upstream repo?

Not if the fork will be creating tags regularly.

File: Formula/libbitcoin-explorer.rb (hunk 2 above)
Comment (ilovezfs) [inline Formula/libbitcoin-explorer.rb]: `dep`, `jose`, `faas-cli`, and `vim` have some good examples.

**Comment by sugarjig** [comment]
@ilovezfs I wrote I slightly more sophisticated test. But now the build fails with this:
```
dyld: Library not loaded: /usr/local/opt/gmp/lib/libgmp.10.dylib
  Referenced from: /usr/local/opt/libbitcoin/lib/libbitcoin.0.dylib
  Reason: image not found
Error: libbitcoin-explorer: failed
```
I'm a little unsure why it's trying to load gmp. When I run `otool` locally, I don't see any reference to gmp:
```
$ otool -L /usr/local/opt/libbitcoin/lib/libbitcoin.0.dylib 
/usr/local/opt/libbitcoin/lib/libbitcoin.0.dylib:
	/usr/local/opt/libbitcoin/lib/libbitcoin.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/boost/lib/libboost_chrono-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_date_time-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_iostreams-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_locale-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_log-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_program_options-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_regex-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_system.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/boost/lib/libboost_thread-mt.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)
	/usr/local/Cellar/libbitcoin/3.3.0/libexec/lib/libsecp256k1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)
```
Any ideas?

**Comment by ilovezfs** [comment]
@sugarjig it's because `libbitcoin` opportunistically decided to link to gmp during https://github.com/Homebrew/homebrew-core/pull/17122.

**Comment by ilovezfs** [comment]
@sugarjig this will fix it: https://github.com/Homebrew/homebrew-core/pull/17220

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@sugarjig I notice in the documentation

```
$ ./install.sh --with-icu --with-png --with-qrencode --build-icu --build-zlib --build-png --build-qrencode --build-boost --prefix=/home/me/myprefix
```
Should we build with those deps too? I believe brew has all of them.

**Comment by sugarjig** [comment]
@ilovezfs I noticed that too, but was going for the bare minimum. I think these flags are for building the core libbitcoin package, so I'd have to go update that formula.

**Comment by ilovezfs** [comment]
@sugarjig no worries. Minimal is good :)

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @sugarjig! Shipped :)

**Comment by sugarjig** [comment]
Great! Thanks for the help!

I doubled checked about those other dependencies. It turns out they are for this package. I'll look into adding `option`s to the formula and submitting a follow up.

**Pull request closed without merging**